core/lib/basic_types/src/url.rs

```diff
@@ -0,0 +1,117 @@
+use std::{fmt, str::FromStr};
+
+use serde::{Deserialize, Deserializer};
+use url::Url;
+
+/// URL with potentially sensitive authentication info (user name and password). Has a specialized
+/// `Debug` implementation to hide sensitive parts and explicit API to expose the underlying URL.
+#[derive(Clone, PartialEq)]
+pub struct SensitiveUrl {
```

Comment:
Not sure in which crate this type should be placed.

Reply:
If we are both to keep it and keep specific, then I think basic types is OK for time being.
If we are to generalize the approach for any kind of potentially sensitive data, then a separate dedicated crate may be better, IMHO.